shared/numberUtils.ts

```diff
@@ -0,0 +1,27 @@
+const num =
+	'zero one two three four five six seven eight nine ten eleven twelve thirteen fourteen fifteen sixteen seventeen eighteen nineteen'.split(
+		' ',
+	);
+const tens = 'twenty thirty forty fifty sixty seventy eighty ninety'.split(' ');
+
+export const number2words = (n: number): string => {
```

Comment:
Would it be worth throwing an error explicitly if a number < 0 is ever passed (and also a non-whole number) to make it clear that's not supported? (Sadly I don't think TS can help here!)

Reply:
hmm, I'm not sure whether to throw an error here or not.

It does look like you can enforce non negative numbers in typescript using generics and template literals like this:

```
type NonNegative = number & { __nonNegative: true };
type AssertPositive<N extends number> = number extends N ? NonNegative : `${N}` extends `-${string}` ? never : N;

function sqrt<N extends number>(n: AssertPositive<N>): number {
    return Math.sqrt(n);
}
```

However, the compiler is not clever enough to do math on the function input .. eg 
`sqrt(5) // this is fine`
`sqrt(-5) // this will give a type error`
`sqrt(5 + 5) // uh ohh, the compiler doesn't know what to do with this and will give a type error`
this can be "fixed" by casting the input using as:
`sqrt(5 + 5 as NonNegative)`
but that's possibly not all that safe as you could just as easily cast a negative number and fool typescript

Reply:
I've settled for now on adding some documentation above the function declaration and adding some out of bounds function input to the unit/jest tests